pkg/admin/transport/handler_user_export_get.go

```diff
@@ -52,11 +58,20 @@ func (h *UserExportGetHandler) handle(w http.ResponseWriter, r *http.Request) er
 		return err
 	}
 
-	response, err := userexport.NewResponseFromTask(task, h.CloudStorage)
+	response, err := userexport.NewResponseFromTask(task)
 	if err != nil {
 		return err
 	}
 
+	// Get presigned download url when the task completed successfully
+	if response.FailedAt == nil {
```

Comment:
I think you need to check the status as well. When the task is still running, FailedAt is nil but it cannot be downloaded yet.

Reply:
added status check